streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -1800,7 +1900,10 @@ public ProductionExceptionHandlerResponse handle(final ErrorHandlerContext conte
                                                          final ProducerRecord<byte[], byte[]> record,
                                                          final Exception exception) {
             assertInputs(context, exception);
-            return response;
+            if (response == null) {
```

Comment:
I don't think the `response` can be null should this be `response.isEmpty`?

Reply:
`null` is in purpose. It's how we setup the code (it's not something the handler return).

In the end `response` is "what should the handler return in this test run", and I missuse it to say "handler should throw an exception". I could also add a new boolean flag `shouldThrowException` instead to not missuse `response` for this case.